### ISSUE TITLE
CBSE Maths grade 7 AI/ML activities

### ISSUE THREAD

**Issue by ndubbaka** [Pragament/coding_simulator_projects#5]
Tools
- https://machinelearningforkids.co.uk/scratch/
- https://playground.raise.mit.edu/create/
- https://quickdraw.withgoogle.com/
- https://experiments.withgoogle.com/thing-translator
- https://soundraw.io/generate_music
- https://research.aimultiple.com/generative-ai-tools/
- https://www.freshbots.org/
- https://appinventor.mit.edu/

**For Class 7 students**, we can introduce slightly more advanced AI/ML concepts while keeping activities hands-on and accessible. Here are **AI/ML activities tailored to Class 7 topics**:

---

## **🧠 AI Concepts for Class 7 Level:**
- **Training vs Testing data**
- **Features and patterns**
- **Simple decision trees**
- **Probability in AI**
- **Basic neural network analogy**

---

### **Activity: "Temperature Prediction Model"**
1. **Collect Data:** 
   - Morning temp (°C) vs Afternoon temp
   - Create table with positive/negative temperatures
2. **"Train Your Model":**
   - Use 10 days of data (training set)
   - Draw scatter plot on graph paper
   - Find relationship: "When morning is X, afternoon is usually Y"
3. **Make Predictions:**
   - Given today's morning temp, predict afternoon
   - Check actual next day (testing)
   - Calculate error (difference)
4. **Discuss:** Why predictions aren't perfect? (Like real AI!)

### **AI Connection:** Weather apps use similar models with more data

---

### **Activity: "Build a Triangle Detective AI"**
1. **Create Decision Tree:**
   ```
   1. All sides equal? → Equilateral
      No → 
   2. Two sides equal? → Isosceles
      No → 
   3. Right angle? → Right-angled
      No → Scalene
   ```
2. **"Code" with Flowchart:** Draw decision flowchart
3. **Test with Cards:** Triangle cards with side lengths/angles
4. **Add Probability:** 
   - "If side1 = side2, 90% chance Isosceles"
   - Track accuracy percentage

### **AI Connection:** Medical diagnosis AI uses similar decision trees

---

### **Activity: "Find the Simplest Form Algorithm"**
1. **Create "AI Rules":**
   - Rule 1: If numerator = denominator → 1
   - Rule 2: If both even → divide by 2
   - Rule 3: Check divisibility by 3, 5, etc.
2. **Manual Execution:** Students follow rules exactly like computer
3. **Efficiency Challenge:** Which rule order is fastest?
4. **Create Flowchart:** Visual algorithm for simplification

### **Extension:** Compare with real AI (Wolfram Alpha fraction simplifier)

---

### **Activity: "Student Grouping by Similarity"**
1. **Collect 2D Data:** 
   - Height (cm) and Math score (%) for 20 students
2. **Plot on Graph:** X=Height, Y=Score
3. **"Find Natural Groups":**
   - Circle clusters visually
   - Name clusters: "Tall High-Scorers", "Short Average", etc.
4. **Calculate Cluster Centers:** Average height & score for each cluster
5. **New Student:** Plot them → Which cluster do they belong to?

### **Real Demo:** Show how Netflix groups similar viewers

---

### **Activity: "Equation Solver Training"**
1. **Train on Patterns:**
   - x + 5 = 10 → solution: x = 10 - 5
   - 2x = 12 → solution: x = 12 ÷ 2
   - x - 3 = 7 → solution: x = 7 + 3
2. **Create Rule Cards:** "If pattern A, apply operation B"
3. **Solve New Equations:** Using rule cards
4. **"AI Error Analysis":** Why 3x + 2 = 11 is harder?

### **Tech Connection:** Show how Photomath app solves equations

---

### **Activity: "Dice Roll Predictor with Confidence"**
1. **Collect Training Data:** Roll dice 60 times, record outcomes
2. **Calculate Probabilities:** 
   - P(1) = count(1)/60, etc.
   - Expected vs actual distribution
3. **Make Predictions with Confidence:**
   - "Next roll will be ≤3 with 50% confidence"
   - "Will be even number with 50% confidence"
   - "Will be 6 with 16.7% confidence"
4. **Compare with Actual:** Track prediction accuracy

### **AI Connection:** Self-driving cars predict "pedestrian will cross with 80% confidence"

---

### **Activity: "Rate the Symmetry"**
1. **Create Symmetry Scoring System:**
   - Perfect mirror = 10 points
   - Small differences = 7 points
   - One side different = 3 points
   - No symmetry = 0 points
2. **"Train" with Examples:** Score known symmetric shapes
3. **Test with New Shapes:** Each student draws, others score
4. **Compare Scores:** How consistent are human "AIs"?

### **Real World:** Facial recognition uses symmetry scores

---

### **Activity: "Discount Price Predictor"**
1. **Dataset Creation:**
   - Original price vs Discount %
   - Final price = Original × (1 - Discount/100)
2. **Plot Graph:** Original price (X) vs Final price (Y)
3. **Draw "AI Line":** Best fit line through points
4. **Predict:** 
   - New item: ₹500 with 20% discount → Predict final price
   - Check with calculation
5. **Profit/Loss Version:** Cost price vs Selling price patterns

---

### **Activity: "Find the Interest Pattern"**
1. **Given Data Table:**
   - Principal | Rate | Time | Interest
   - ₹1000 | 5% | 2 years | ₹100
   - ₹2000 | 10% | 1 year | ₹200
   - etc.
2. **"AI Discovery Task":** Find the hidden formula
3. **Test Hypothesis:** Try your formula on new examples
4. **Compare:** SI = P×R×T/100
5. **Discussion:** How did you "learn" the formula? (Like AI!)

---

### **Activity: "Build a Proof Generator"**
1. **Create Rule Set:**
   - If 3 sides equal → SSS congruence
   - If 2 sides & included angle equal → SAS
   - etc.
2. **Given Problem:** "Prove △ABC ≅ △DEF"
3. **Check Conditions:** 
   - AB = DE? ✓
   - BC = EF? ✓
   - ∠B = ∠E? ✓
   → Therefore SAS congruence
4. **Flowchart Maker:** Create decision diagram for any triangle pair

---

## **🛠️ Digital Tools for Class 7:**

1. **Machine Learning for Kids** (more advanced projects)
2. **Google's Teachable Machine** (image/sound classification)
3. **Scratch with ML Extension** (block coding)
4. **Orange Data Mining** (visual programming - simplified)
5. **AI Playground** (pre-trained models to play with)

---

### **Project: "Smart Study Time Predictor"**
1. **Problem:** How much time needed for math homework?
2. **Collect Data:** Topic difficulty (1-5 rating) vs Time taken
3. **Create Model:** Predict time for new topics
4. **Test & Refine:** Compare predictions with actual
5. **Present:** As an "AI app" proposal

### **Project: "Quiz Question Difficulty Classifier"**
1. **Collect Questions:** Label easy/medium/hard
2. **Find Features:** Word count, operations needed, steps required
3. **Create Rules:** If word count > 30 AND steps > 3 → probably hard
4. **Test:** On new questions

---

### **1. "Design an AI Math Tutor"**
- What features would it have?
- How would it learn your weak areas?
- Draw interface
- Write "training process"

### **2. "AI vs Human Math Competition"**
- Simple calculations
- Pattern recognition
- Probability predictions
- Compare speed & accuracy

### **3. "Bias Detective in AI"**
- If AI trained only on boys' math scores...
- What would it predict for girls?
- Importance of diverse training data

---

## **📊 Assessment Ideas:**

1. **"Explain to a Friend"**: How an AI email filter works using math concepts
2. **"Build a Simple Classifier"**: For types of triangles or numbers
3. **"Error Analysis"**: Why an AI might confuse ⅓ and ¼
4. **"Future AI Idea"**: How AI could help learn their hardest math topic

---

### **Neural Network Analogy: "Team of Specialists"**
- Each student = one "neuron"
- First row: "Feature detectors" (finds numbers, operations)
- Second row: "Pattern recognizers" (sees it's a fraction problem)
- Third row: "Solvers" (applies fraction rules)
- Final output: Answer

### **Training Process:**
1. **Try** (make prediction)
2. **Check** (compare with answer)
3. **Adjust** (learn from mistake)
4. **Repeat**

---

## **✅ Key Learning Outcomes:**

1. **AI makes predictions based on patterns**
2. **More and diverse data = better AI**
3. **AI can handle complex decisions through simple steps**
4. **Probability helps AI express uncertainty**
5. **They can design simple AI systems for math problems**

---

## **🌟 Quick Start Activities (45 minutes each):**

**Week 1:** Activity 2 (Triangle Classifier) - visual, concrete  
**Week 2:** Activity 6 (Probability Predictor) - connects to statistics  
**Week 3:** Activity 8 (Percentage Predictor) - real-world application  
**Week 4:** Project - "Smart Study Time Predictor"

These activities bridge **Class 7 math curriculum** with **foundational AI concepts** in an engaging, age-appropriate way!
